clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1253,6 +1263,46 @@ class SyclKernelBodyCreator
   }
 
   void leaveStruct(const CXXRecordDecl *, FieldDecl *FD) final {
```

Comment:
This and function below for base class needs to be refactored. I notice some discrepancy in code also. I am taking a look and will upload new patch in a bit